test/grouping.jl

```diff
@@ -1,8 +1,33 @@
 module TestGrouping
 
-using Test, DataFrames, Random, Statistics
+using Test, DataFrames, Random, Statistics, PooledArrays
 const ≅ = isequal
 
+"""Check that groups in gd are equal to provided data frames, ignoring order"""
+function isequal_unordered(gd::GroupedDataFrame,
+                            dfs::AbstractVector{<:AbstractDataFrame})
+    n = length(gd)
+    @assert n == length(dfs)
+    remaining = Set(1:n)
+    for i in 1:n
+        for j in remaining
+            if gd[i] ≅ dfs[j]
+                pop!(remaining, j)
```

Comment:
does not this `pop!` potentially break iteration (I think it is OK, but just asking to be sure if you checked - some 1 year ago I was fixing something around this in Julia Base, but cannot recall in detail)

Reply:
Actually I can see it is wrong. Run the following code several times
```
function test()
    a = rand(1:1000, 20)
    x = Set(a)
    y = Int[]
    i = 0
    for v in x
        i += 1
        push!(y, pop!(x, v))
    end

    return isempty(x), i, a, y
end
```

and you will see that sometimes `i<20` and then `a` and `y` are not the same.

Reply:
And now I think I remember - you can safely delete elements, but you cannot add elements while iterating.

Reply:
Good point. I'm not sure what the rules are, but here I think the fact that we call `break` immediately should prevent any problems anyway.

Reply:
True. In general I have an Pavlovian conditioning from other languages when I see that a collection that is iterated over is mutated inside the iterator 😄.